pkg/agent/filestore/filestore.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2025 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestore
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+)
+
+// Set it to NewMemMapFs as the file system may be not writable.
+// Change it to NewOsFs to evaluate performance when writing to disk.
+// It's for testing only.
+var newFS = afero.NewMemMapFs
+
+type AnyObjectWithUID struct {
+	UID    string
+	Object any
+}
+
+type Serializer interface {
+	Encode(obj any, w io.Writer) error
+	Decode(data []byte, into *AnyObjectWithUID) error
+}
+
+type GobSerializer struct{}
+
+func (j *GobSerializer) Encode(obj any, w io.Writer) error {
+	return gob.NewEncoder(w).Encode(obj)
+}
+
+func (j *GobSerializer) Decode(data []byte, into *AnyObjectWithUID) error {
+	dec := gob.NewDecoder(bytes.NewReader(data))
+	return dec.Decode(into)
+}
+
+func NewFakeFileStore() *FileStore {
+	gob.Register(&interfacestore.SRIOVInterfaceConfig{})
+	serializer := &GobSerializer{}
+	fs := afero.NewBasePathFs(newFS(), "/var/run/antrea-test/file-store")
+	s, _ := NewFileStore(fs, "sriov", serializer)
+	return s
+}
+
+// FileStore encodes and stores golang objects in files. Each object will be stored in a
+// separate file under the given directory.
+type FileStore struct {
```

Comment:
this seem similar to the code that @tnqn added to save / restore policy objects in agent: `pkg/agent/controller/networkpolicy/filestore.go`

I'm fairly confident we can move that code to `pkg/agent/util/filestore` and share it

Reply:
Can we just use code gen and have `SRIOVInterfaceConfig` meet the requirements?
```go
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
type SRIOVInterfaceConfig {

}
```

We already use code gen for some interface store types: 
https://github.com/antrea-io/antrea/blob/93378659448b21cddd5d97f2863fa65d3edc9e38/pkg/agent/interfacestore/types.go#L102-L103

It's better than the code duplication IMO.

cc @tnqn